python/lsst/ap/association/assoc_afw_utils.py

```diff
@@ -0,0 +1,279 @@
+#
+# LSST Data Management System
+# Copyright 2017 LSST/AURA.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+from collections import OrderedDict as oDict
+
+import numpy as np
+
+import lsst.afw.table as afwTable
+
+__all__ = ["make_minimal_dia_object_schema",
+           "make_minimal_dia_source_schema",
+           "ccdVisitSchema"]
+
+
+"""Defines afw schemas and conversions for use in ap_association tasks.
+
+Until a more finalized interface between the prompt products database (PPD) can
+be established, we put many utility functions for converting `lsst.afw.table` and
+`lsst.afw.image` objects to a PPD. This includes both mapping schemas between
+different catalogs and the DB.
+"""
+
+
+def make_minimal_dia_object_schema(filter_names=[]):
```

Comment:
Don't default to an empty list (or any mutable). Use `None` instead, if it's optional, and set it to a list at the start of the method.

http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments